packages/swapper/src/swappers/ChainflipSwapper/swapperApi/getTradeQuote.ts

```diff
@@ -0,0 +1,330 @@
+import { getUtxoTxFees } from '../utils/getUtxoTxFees'
+import { isSupportedAssetId, isSupportedChainId } from '../utils/helpers'
+
+export const getTradeQuote = async (
+  input: GetTradeQuoteInput,
+  deps: SwapperDeps,
+): Promise<Result<TradeQuote[], SwapErrorRight>> => {
+  const {
+    sellAsset,
+    buyAsset,
+    accountNumber,
+    receiveAddress,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmount,
+    affiliateBps: commissionBps,
+  } = input
+
+  if (accountNumber === undefined) {
+    return Err(
+      makeSwapErrorRight({
+        message: `accountNumber is required`,
+        code: TradeQuoteError.UnknownError,
+      }),
+    )
+  }
+
+  if (!isSupportedChainId(sellAsset.chainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  if (!isSupportedChainId(buyAsset.chainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  if (!isSupportedAssetId(sellAsset.chainId, sellAsset.assetId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${sellAsset.name}' on chainId '${sellAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+        details: { chainId: sellAsset.chainId, assetId: sellAsset.assetId },
+      }),
+    )
+  }
+
+  if (!isSupportedAssetId(buyAsset.chainId, buyAsset.assetId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${buyAsset.name}' on chainId '${buyAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+        details: { chainId: buyAsset.chainId, assetId: buyAsset.assetId },
+      }),
+    )
+  }
+
+  const sellChainflipChainKey = `${sellAsset.symbol.toLowerCase()}.${
+    chainIdToChainflipNetwork[sellAsset.chainId]
+  }`
+  const buyChainflipChainKey = `${buyAsset.symbol.toLowerCase()}.${
+    chainIdToChainflipNetwork[buyAsset.chainId]
+  }`
+
+  const brokerUrl = deps.config.REACT_APP_CHAINFLIP_API_URL
+  const apiKey = deps.config.REACT_APP_CHAINFLIP_API_KEY
+
+  // Subtract the BaaS fee to end up at the final displayed commissionBps
+  let serviceCommission = parseInt(commissionBps) - CHAINFLIP_BAAS_COMMISSION
+  if (serviceCommission < 0) serviceCommission = 0
+
+  const maybeQuoteResponse = await chainflipService.get<ChainflipBaasQuoteQuote[]>(
+    `${brokerUrl}/quotes-native` +
+      `?apiKey=${apiKey}` +
+      `&sourceAsset=${sellChainflipChainKey}` +
+      `&destinationAsset=${buyChainflipChainKey}` +
+      `&amount=${sellAmount}` +
+      `&commissionBps=${serviceCommission}`,
+  )
+
+  if (maybeQuoteResponse.isErr()) {
+    const error = maybeQuoteResponse.unwrapErr()
+    const cause = error.cause as AxiosError<any, any>
+
+    if (
```

Comment:
This can probably be done cleaner, the error responses follow the problem details rfc, maybe axios has some way to properly parse that to a typed thing instead of the `any` I had to do here

Reply:
Also, minimum amounts are also on the https://chainflip-broker.io/assets endpoint as `minimalAmountNative`